hydra-cluster/test/Test/HydraExplorerSpec.hs

```diff
@@ -0,0 +1,239 @@
+            seedFromFaucet_ cardanoNode aliceCardanoVk 100_000_000 (contramap FromFaucet tracer)
+            aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            aliceHeadId <- withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] initHead
+
+            (bobCardanoVk, _bobCardanoSk) <- keysFor Bob
+            seedFromFaucet_ cardanoNode bobCardanoVk 100_000_000 (contramap FromFaucet tracer)
+            bobChainConfig <- chainConfigFor Bob tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            bobHeadId <- withHydraNode hydraTracer bobChainConfig tmpDir 2 bobSk [] [2] initHead
+
+            withHydraExplorer cardanoNode $ \explorer -> do
+              headExplorerSees explorer "HeadInitTx" aliceHeadId
+              headExplorerSees explorer "HeadInitTx" bobHeadId
+
+  it "can query for all hydra heads observed" $
+    failAfter 60 $
+      showLogsOnFailure "HydraExplorerSpec" $ \tracer -> do
+        withTempDir "hydra-explorer-get-heads" $ \tmpDir -> do
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \cardanoNode@RunningNode{nodeSocket} -> do
+            let hydraTracer = contramap FromHydraNode tracer
+            hydraScriptsTxId <- publishHydraScriptsAs cardanoNode Faucet
+            withHydraExplorer cardanoNode $ \explorer -> do
+              (aliceCardanoVk, _aliceCardanoSk) <- keysFor Alice
+              aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+              aliceHeadId <- withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] $ \hydraNode -> do
+                seedFromFaucet_ cardanoNode aliceCardanoVk 100_000_000 (contramap FromFaucet tracer)
+
+                send hydraNode $ input "Init" []
+
+                aliceHeadId <- waitMatch 5 hydraNode $ \v -> do
+                  guard $ v ^? key "tag" == Just "HeadIsInitializing"
+                  v ^? key "headId" . _String
+
+                headExplorerSees explorer "HeadInitTx" aliceHeadId
+
+                pure aliceHeadId
+
+              (bobCardanoVk, _bobCardanoSk) <- keysFor Bob
+              bobChainConfig <- chainConfigFor Bob tmpDir nodeSocket hydraScriptsTxId [] cperiod
+              bobHeadId <- withHydraNode hydraTracer bobChainConfig tmpDir 2 bobSk [] [2] $ \hydraNode -> do
+                seedFromFaucet_ cardanoNode bobCardanoVk 100_000_000 (contramap FromFaucet tracer)
+
+                send hydraNode $ input "Init" []
+
+                bobHeadId <- waitMatch 5 hydraNode $ \v -> do
+                  guard $ v ^? key "tag" == Just "HeadIsInitializing"
+                  v ^? key "headId" . _String
+
+                headExplorerSees explorer "HeadInitTx" bobHeadId
+
+                send hydraNode $ input "Abort" []
+
+                headExplorerSees explorer "HeadAbortTx" bobHeadId
+
+                pure bobHeadId
+
+              manager <- HTTPClient.newTlsManager
```

Comment:
I don't think you need to use tls here, we don't hit `https` endpoint. Also there is a  `parseRequestThrow` already in scope so you might want to use `Network.HTTP.Simple` and then we don't need extra dependencies added in `hydra-cluster.cabal` file.

Reply:
fyi: this does requires me to add `http-conduit` as dependency